lib/src/test/java/io/ably/lib/http/HttpTest.java

```diff
@@ -0,0 +1,637 @@
+package io.ably.lib.http;
+
+import fi.iki.elonen.NanoHTTPD;
+import fi.iki.elonen.router.RouterNanoHTTPD;
+import io.ably.lib.test.util.StatusHandler;
+import io.ably.lib.transport.Defaults;
+import io.ably.lib.types.AblyException;
+import io.ably.lib.types.ClientOptions;
+import io.ably.lib.types.ErrorInfo;
+import io.ably.lib.types.Param;
+import org.hamcrest.Description;
+import org.hamcrest.TypeSafeMatcher;
+import org.junit.*;
+import org.junit.rules.ExpectedException;
+import org.mockito.ArgumentCaptor;
+import org.mockito.Mockito;
+import org.mockito.invocation.InvocationOnMock;
+import org.mockito.stubbing.Answer;
+
+import java.io.IOException;
+import java.net.MalformedURLException;
+import java.net.URL;
+import java.util.ArrayList;
+import java.util.List;
+
+import static org.hamcrest.Matchers.*;
+import static org.junit.Assert.assertThat;
+import static org.mockito.AdditionalMatchers.aryEq;
+import static org.mockito.Matchers.any;
+import static org.mockito.Matchers.anyBoolean;
+import static org.mockito.Matchers.anyString;
+import static org.mockito.Mockito.*;
+
+/**
+ * Created by gokhanbarisaker on 2/2/16.
+ */
+public class HttpTest {
+
+	private static final String PATTERN_HOST_FALLBACK = "(?i)[a-e]\\.ably-realtime.com";
+
+	@Rule
+	public ExpectedException thrown = ExpectedException.none();
+	private static RouterNanoHTTPD server;
+
+
+	@BeforeClass
+	public static void setUp() throws IOException {
+		server = new RouterNanoHTTPD(27331);
+		server.addRoute("/status/:code", StatusHandler.class);
+		server.start(NanoHTTPD.SOCKET_READ_TIMEOUT, true);
+
+		while (!server.wasStarted()) {
+			try {
+				Thread.sleep(100);
+			} catch (InterruptedException e) {
+				e.printStackTrace();
+			}
+		}
+	}
+
+	@AfterClass
+	public static void tearDown() {
+		server.stop();
+	}
+
+
+	/*******************************************
+	 * Spec: RSC15
+	 *******************************************/
+
+	/**
+	 * <p>
+	 * Validates {@code Http} performs fallback behavior httpMaxRetryCount number of times at most,
+	 * when host & fallback hosts are unreachable. Then, finally throws an error.
+	 * </p>
+	 * <p>
+	 * Spec: RSC15a
+	 * </p>
+	 *
+	 * @throws Exception
+	 */
+	@Test
+	public void http_ably_execute_fallback() throws AblyException {
```

Comment:
Could we add another integration test that confirms fallbacks are not used when the default host is not used?